src/target/cortexm.c

```diff
@@ -506,7 +506,7 @@ bool cortexm_attach(target *t)
 	priv->flash_patch_revision = (r >> 28);
 	priv->hw_watchpoint_max = CORTEXM_MAX_WATCHPOINTS;
 	r = target_mem_read32(t, CORTEXM_DWT_CTRL);
-	if ((r >> 28) > priv->hw_watchpoint_max)
+	if ((r >> 28) < priv->hw_watchpoint_max)
```

Comment:
Please describe the impacts of this change as that seems pretty radical - a before-and-after of what `priv->hw_watchpoint_max` looks like would be very helpful just to get a feel for this.

Reply:
There were 2 issues with this bug. For targets with less than 4 watchpoints, it made GDB think there were actually 4 watchpoints available. For targets with more than 4 watchpoints, it caused memory corruption in BMP.
hw_watchpoints is a static array with size 4. This condition was meant to clamp the number of watchpoints to that limit, but due to the typo it didn't do that.

Reply:
Ok, that makes this seem much more reasonable. Thank you for the explanation!

Reply:
As all CortexM has 4 watchpoints, we never hit the issue. But your finding is right. For the FPU registers, i guess few users looked at the FPU registers in the debugger and so noone noticed, but your fix is right too.